src/grain/createGrainAllocation.test.js

```diff
@@ -0,0 +1,356 @@
+    return receipts.map(({address, amount}) => ({
+      address,
+      amount: fmt(amount),
+    }));
+  }
+  function safeAllocation(allocation: GrainAllocationV1) {
+    return {
+      ...allocation,
+      budget: fmt(allocation.budget),
+      receipts: safeReceipts(allocation.receipts),
+    };
+  }
+  function expectAllocationsEqual(h1, h2) {
+    expect(safeAllocation(h1)).toEqual(safeAllocation(h2));
+  }
+
+  const immediateStrategy: ImmediateV1 = deepFreeze({
+    type: "IMMEDIATE",
+    version: 1,
+  });
+
+  const balancedStrategy: BalancedV1 = deepFreeze({
+    type: "BALANCED",
+    version: 1,
+  });
+
+  describe.each([[immediateStrategy], [balancedStrategy]])(
+    "Common Tests for %o",
+    (strategy: AllocationStrategy) => {
+      const BUDGET = ONE;
+
+      it("throws an error if given an unsupported strategy", () => {
+        const unsupportedStrategy = {
+          ...strategy,
+          version: 2,
+        };
+        expect(() =>
+          createGrainAllocation(
+            unsupportedStrategy,
+            BUDGET,
+            credHistory,
+            new Map()
+          )
+        ).toThrowError(`Unsupported ${strategy.type} version: 2`);
+      });
+
+      it("throws an error if given an unsupported strategy and an empty credHistory", () => {
+        const unsupportedStrategy = {
+          ...strategy,
+          version: 2,
+        };
+        expect(() =>
+          createGrainAllocation(unsupportedStrategy, BUDGET, [], new Map())
+        ).toThrowError(`Unsupported ${strategy.type} version: 2`);
+      });
+
+      describe("it should return an empty allocation when", () => {
+        const emptyAllocation = deepFreeze({
+          version: GRAIN_ALLOCATION_VERSION_1,
+          strategy,
+          budget: BUDGET,
+          receipts: [],
+        });
+
+        it("the budget is zero", () => {
+          const actual = createGrainAllocation(
+            strategy,
+            ZERO,
+            credHistory,
+            new Map()
+          );
+
+          expectAllocationsEqual(actual, {
+            ...emptyAllocation,
+            budget: ZERO,
+          });
+        });
+
+        it("there are no cred scores at all", () => {
+          const actual = createGrainAllocation(strategy, BUDGET, [], new Map());
+          expectAllocationsEqual(actual, emptyAllocation);
+        });
+
+        it("all the cred sums to 0", () => {
+          const actual = createGrainAllocation(
+            strategy,
+            BUDGET,
+            [
+              {
+                intervalEndMs: 500,
+                cred: new Map([
+                  [foo, 0],
+                  [bar, 0],
+                ]),
+              },
+            ],
+            new Map()
+          );
+
+          expectAllocationsEqual(actual, emptyAllocation);
+        });
+      });
+    }
+  );
+
+  describe("immediateAllocation", () => {
+    const BUDGET = ONE;
```

Comment:
Nit: later on, we're using statements like `ONE / 2n` for the expected result, rather than `BUDGET / 2n`.
Suggest either using `BUDGET` or `ONE` consistently.

Reply:
done